File: 2644.java
class Solution {
    public int maxDivScore(int[] nums, int[] divisors) {
        int[] scores = new int[divisors.length];
        for (int i = 0; i < nums.length; ++i) {
            for (int j = 0; j < divisors.length; ++j) {
                if (nums[i] % divisors[j] == 0) {
                    scores[j] += 1;
                }
            }
        }
        int max_idx = 0;
        for (int i = 1; i < scores.length; ++i) {
            if (scores[i] > scores[max_idx]) {
                max_idx = i;
            }
            else if (scores[i] == scores[max_idx] && divisors[i] < divisors[max_idx]) {
                max_idx = i;
            }
        }
        return divisors[max_idx];
    }
}

File: 2644.rs
use std::collections::{binary_heap, HashSet};
use std::process::id;
use std::sync::TryLockError;
use std::thread::current;
use std::{default, string, vec};
use std::{collections::HashMap, hash::Hash, collections::BinaryHeap};
use std::cmp::{Ord, Ordering};

impl Solution {
    pub fn max_div_score(nums: Vec<i32>, divisors: Vec<i32>) -> i32 {
        let mut max_idx: usize = 0;
        let mut max_score: i32 = -1;
        for j in 0..divisors.len() {
            let mut score = 0;
            for i in 0..nums.len() {
                if nums[i] % divisors[j] == 0 {
                    score += 1;
                }
            }
            if score > max_score || (score == max_score && divisors[j] < divisors[max_idx]) {
                max_idx = j;
                max_score = score;
            }
        }
        divisors[max_idx]
    }
}